File: App/ClientApp/app/Room.ts
import { Technique } from "./technique";
import { TypeTechnique } from "./TypeTechnique";
import { techniqueAndType } from "./techniquAndType";

export class Room {
    constructor(
        public id?: number,
        public length?: number,
        public width?: number,
        public name?: string,
        public technique?: techniqueAndType []
    ) { }

}

File: App/Models/Room.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace App.Models
{
    public class Room
    {
        public int Id { get; set; }
        public double Length { get; set; }
        public double Width { get; set; }
        public string Name { get; set; }
        public TechniqueAndType[] Technique { get; set; }
    }
}
